src/main/java/it/uniroma2/pellegrini/z64sim/view/MainWindow.java

```diff
@@ -100,6 +102,46 @@ public void keyTyped(KeyEvent e) {
         runButton.addActionListener(actionEvent -> {
             SimulatorController.run();
         });
+
+        mainPanel.registerKeyboardAction(
+            e -> this.saveFile(),
+            KeyStroke.getKeyStroke(KeyEvent.VK_S, InputEvent.CTRL_DOWN_MASK),
```

Comment:
This is not nice to Mac users, as there is no `ctrl` button there.

Java guidelines suggest to rely on the platform's toolkit in `java.awt.Toolkit`, in particular the `getMenuShortcutKeyMaskEx()` method, for this purpose.